widgets/search-box/search-box.js

```diff
@@ -100,13 +100,28 @@ function searchBox({
 
       // Add all the needed attributes and listeners to the input
       this.addDefaultAttributesToInput(input, initialState.query);
+
+      // Keep keyup to handle searchOnEnterKeyPressOnly
       input.addEventListener('keyup', (e) => {
         helper.setQuery(input.value);
-        if (!searchOnEnterKeyPressOnly || e.keyCode === KEY_ENTER) {
+        if (searchOnEnterKeyPressOnly && e.keyCode === KEY_ENTER) {
           helper.search();
         }
       });
 
+      function inputCallback(e) {
+        helper.setQuery(e.currentTarget.value);
+        if (!searchOnEnterKeyPressOnly) {
+          helper.search();
+        }
+      }
+
+      if (window.addEventListener) {
```

Comment:
Can you reverse the test to (if attachEvent) else addeventlistener?

I double checked and it will support 9. IE9 has both addEventListener and attachEvent but the "input" event in IE9 will not be triggered when deleting text. While onpropertychange will. And propertychange must be done on attachEvent.

Double check that it works with a real IE9/10 (which have both attachEvent and addEventListener) because this is a tricky change.

Also check that it does not trigger too much the event, it seems that "propertychange" is triggered for every property change on the input. **tricky** :)